[BUG] api build fails on node-fetch ESM import

### Which part of SahiDawa is broken?
Backend — Express API (apps/api)

### Bug Description
The monorepo build currently fails on `main` before unrelated web PRs can finish CI because `apps/api/src/routes/medicine.ts` imports `node-fetch` from a CommonJS TypeScript module.

Evidence from `origin/main`:
- `apps/api/src/routes/medicine.ts:4` uses `import fetch from "node-fetch";`

CI/local failure excerpt:
```text
src/routes/medicine.ts(4,19): error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("node-fetch")' call instead.
```

### Expected Behavior
`npm run build` should build the API workspace without ESM/CommonJS interop errors.

### Screenshots
Not applicable. Terminal proof above.

### GSSoC 2026
- [x] I am a GSSoC 2026 participant and would like to fix this bug.

### Code of Conduct
- [x] I have read and agree to SahiDawa's Code of Conduct.

--- a/apps/api/src/routes/medicine.ts
+++ b/apps/api/src/routes/medicine.ts
@@ -1,7 +1,5 @@
 import { Router, Request, Response } from "express";
 import multer from "multer";
-import FormData from "form-data";
-import fetch from "node-fetch";
 import { createClient } from "redis";
 import { supabase } from "../db/client";
 import { scanQueryLimiter } from "../middleware/rateLimit";
@@ -47,17 +45,14 @@ router.post(
                 return res.status(400).json({ success: false, error: "No audio file provided." });
             }
 
-            // Forward audio to Python FastAPI ML service
             const form = new FormData();
-            form.append("audio", req.file.buffer, {
-                filename: "recording.webm",
-                contentType: req.file.mimetype,
-            });
+            const audioBytes = Uint8Array.from(req.file.buffer);
+            const audioBlob = new Blob([audioBytes], { type: req.file.mimetype });
+            form.append("audio", audioBlob, "recording.webm");
 
             const mlResponse = await fetch(`${ML_SERVICE_URL}/voice/verify`, {
                 method: "POST",
                 body: form,
-                headers: form.getHeaders(),
             });
 
             if (!mlResponse.ok) {

--- a/apps/web/components/VoiceVerify.tsx
+++ b/apps/web/components/VoiceVerify.tsx
@@ -104,7 +104,7 @@ export default function VoiceVerify() {
       recorder.start();
       mediaRecorderRef.current = recorder;
       setIsRecording(true);
-    } catch (err) {
+    } catch {
       setError("Microphone access denied. Please allow microphone access and try again.");
     }
   }, []);

--- a/apps/web/components/medicine/MedicineSafetyPanel.tsx
+++ b/apps/web/components/medicine/MedicineSafetyPanel.tsx
@@ -29,14 +29,8 @@ import {
     TriangleAlert,
     Refrigerator,
 } from "lucide-react";
-import {
-    SAFETY_DISCLAIMER,
-    type MedicineSafetyProfile,
-    type AgeGroup,
-    type DietaryRule,
-} from "./MedicineSafetyData";
+import { type MedicineSafetyProfile, type AgeGroup } from "./MedicineSafetyData";
 import { fetchSafetyProfile } from "@/lib/medicineSafetyService";
-import { getSafetyProfile, type MedicineSafetyProfile, type AgeGroup } from "./MedicineSafetyData";
 
 // ── Props ─────────────────────────────────────────────────────────────────────
 type MedicineSafetyPanelProps = {
@@ -100,8 +94,6 @@ export function MedicineSafetyPanel({ searchQuery, onClose }: MedicineSafetyPane
             cancelled = true;
         };
     }, [searchQuery]);
-
-    const t = useTranslations("medicineSafety");
 
     const t = useTranslations("medicineSafety");
 

--- a/apps/web/tests/MedicineSearchSelect.test.tsx
+++ b/apps/web/tests/MedicineSearchSelect.test.tsx
@@ -2,7 +2,7 @@
  * @jest-environment jsdom
  */
 import "@testing-library/jest-dom";
-import { render, screen, waitFor } from "@testing-library/react";
+import { act, render, screen, waitFor } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 // import { describe, test, expect, vi, beforeEach, afterEach } from "vitest";
 import MedicineSearchSelect from "../src/components/MedicineSearchSelect";
@@ -66,7 +66,7 @@ describe("MedicineSearchSelect", () => {
     });
 
     test("clear button calls onChange with null", async () => {
-        const user = userEvent.setup();
+        const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
 
         const onChange = jest.fn();
 
@@ -106,7 +106,9 @@ describe("MedicineSearchSelect", () => {
 
         await user.type(input, "cro");
 
-        jest.advanceTimersByTime(300);
+        await act(async () => {
+            jest.advanceTimersByTime(300);
+        });
 
         await waitFor(() => {
             expect(onSearch).toHaveBeenCalled();
@@ -166,7 +168,7 @@ describe("MedicineSearchSelect", () => {
     });
 
     test("clear history removes history items", async () => {
-        const user = userEvent.setup();
+        const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
 
         localStorage.setItem(
             "sahidawa_search_history",

--- a/apps/web/tests/profile-page.test.tsx
+++ b/apps/web/tests/profile-page.test.tsx
@@ -21,7 +21,7 @@ describe("ProfilePage navigation and guest state", () => {
     it("renders a back-to-home link pointing at the localized home route", () => {
         render(<ProfilePage />);
 
-        const backLink = screen.getByRole("link", { name: /back to home/i });
+        const backLink = screen.getByRole("link", { name: /go back to previous page/i });
         expect(backLink).toHaveAttribute("href", "/");
     });
 

